src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplateSupport.java

```diff
@@ -16,23 +16,41 @@
 
 package org.springframework.data.couchbase.core;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
 import org.springframework.data.couchbase.core.convert.CouchbaseConverter;
 import org.springframework.data.couchbase.core.convert.translation.JacksonTranslationService;
 import org.springframework.data.couchbase.core.convert.translation.TranslationService;
 import org.springframework.data.couchbase.core.mapping.CouchbaseDocument;
 import org.springframework.data.couchbase.core.mapping.CouchbasePersistentEntity;
 import org.springframework.data.couchbase.core.mapping.CouchbasePersistentProperty;
+import org.springframework.data.couchbase.core.mapping.event.*;
 import org.springframework.data.couchbase.repository.support.MappingCouchbaseEntityInformation;
 import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.callback.EntityCallbacks;
 import org.springframework.data.mapping.context.MappingContext;
 import org.springframework.data.mapping.model.ConvertingPropertyAccessor;
+import org.springframework.util.Assert;
 
-public class CouchbaseTemplateSupport {
+/**
+ * encode/decode support for CouchbaseTemplate
+ *
+ * @author Michael Nitschinger
+ * @author Michael Reiche
+ * @since 3.0
+ */
+public class CouchbaseTemplateSupport implements ApplicationContextAware {
+	private static final Logger LOG = LoggerFactory.getLogger(CouchbaseTemplate.class);
```

Comment:
wrong logger class name

Reply:
I used CouchbaseTemplate as it does most of the work for CouchbaseTemplate, and if someone wanted to trace/log/debug CouchbaseTemplate, they would certainly want to see what was going on in CouchbaseTemplateSupport.